apps/prairielearn/assets/scripts/lib/questionsTable.ts

```diff
@@ -9,7 +9,36 @@ import { AssessmentBadge } from '../../../src/components/AssessmentBadge.html.js
 import { SyncProblemButton } from '../../../src/components/SyncProblemButton.html.js';
 import { TagBadgeList } from '../../../src/components/TagBadge.html.js';
 import { TopicBadge } from '../../../src/components/TopicBadge.html.js';
+import { type Topic } from '../../../src/lib/db-types.js';
+import { type QuestionsPageData } from '../../../src/models/questions.js';
 
+import { type ExtendedBootstrapTableOptions } from './bootstrapTable.js';
+
+type PrefixSuffixObjectKeys<T extends Record<string, any>, P extends string, S extends string> = {
+  [K in keyof T as K extends string ? `${P}${K}${S}` : never]: T[K];
+};
+
+type AssessmentGlobals = PrefixSuffixObjectKeys<Record<string, any>, 'assessments', 'Formatter'> &
+  PrefixSuffixObjectKeys<Record<string, any>, 'assessments', 'List'>;
+
+declare global {
+  interface Window extends AssessmentGlobals {
+    topicList: () => any;
+    tagsList: () => any;
+    sharingSetsList: () => any;
+    versionList: () => any;
+    qidFormatter: (_qid: any, question: QuestionsPageData) => any;
+    topicFormatter: (_topic: any, question: QuestionsPageData) => any;
+    tagsFormatter: (_tags: any, question: QuestionsPageData) => any;
+    sharingSetFormatter: (_sharing_sets: any, question: QuestionsPageData) => any;
+    versionFormatter: (_version: any, question: QuestionsPageData) => any;
+    topicSorter: (topicA: Topic, topicB: Topic) => any;
+    genericFilterSearch: (search: string, value: string) => any;
+    badgeFilterSearch: (search: string, value: string) => any;
+  }
+}
+
+// TODO: Pull in correct type from env types
```

Comment:
The correct type for what exactly?

Reply:
Good catch -- this is a stale comment about QuestionsPageData.